src/app/shared/disabled-directive.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  Directive,
+  HostBinding,
+  HostListener,
+  Input,
+} from '@angular/core';
+
+@Directive({
+  selector: '[dsDisabled]',
+  standalone: true,
+})
+export class DisabledDirective {
+
+    @Input() set dsDisabled(value: boolean) {
+    this.isDisabled = !!value;
+  }
+
+    @HostBinding('attr.aria-disabled') isDisabled = false;
+    @HostBinding('class.disabled') get disabledClass() { return this.isDisabled; }
+
+    @HostListener('click', ['$event'])
+    handleClick(event: Event) {
```

Comment:
It would also be good to briefly describe the purpose of the methods in this class.  I realize they are just trying to block the ability to click or interact with the element, but we need to describe that so that other developers are aware of what this directive is doing.